jaxrs/src/main/java/org/killbill/billing/jaxrs/resources/JaxRsResourceBase.java

```diff
@@ -533,4 +535,48 @@ protected void verifyNumberOfElements(int actual, int expected, String errorMess
         Preconditions.checkArgument(actual == expected, errorMessage);
     }
 
+    protected Response createPaymentResponse(final UriInfo uriInfo, final Payment payment, final TransactionType transactionType, @Nullable final String transactionExternalKey) {
+        final PaymentTransaction createdTransaction = findCreatedTransaction(payment, transactionType, transactionExternalKey);
+        Preconditions.checkNotNull(createdTransaction, "No transaction of type '%s' found", transactionType);
```

Comment:
Yeah, that is still broken, but now is broken uniformly :)

We could return a No Content response, but I'm not sure how helpful would that be without some hint on why the payment was aborted. Also, it's impossible to tell if the payment is null b/c it was aborted, or b/c some regression was introduced into code.

I believe throwing a NPE is probably the best (or least worst) way to go, since we don't ever expect aborted payments right now, and if it happens it will be easy it to notice.